### PR TITLE
Fork ResizeObserver polyfill to support shadow DOM and FIE

### PR DESCRIPTION
**Pull request by dvoytenko**
Partial for #31540.

See https://github.com/que-etc/resize-observer-polyfill/pull/79 for more details.

### CODE REVIEW

**Comment by samouri** [comment]
nit: rename from `upgrade` to `fork` ?

File: build-system/tasks/update-packages.js [hunk 1]
```diff
@@ -104,10 +104,10 @@ function patchIntersectionObserver() {
 function patchResizeObserver() {
   // Copies intersection-observer into a new file that has an export.
   const patchedName =
-    'node_modules/resize-observer-polyfill/resize-observer.install.js';
+    'node_modules/resize-observer-polyfill/ResizeObserver.install.js';
   let file = fs
     .readFileSync(
-      'node_modules/resize-observer-polyfill/dist/ResizeObserver.global.js'
```
Comment (samouri) [inline build-system/tasks/update-packages.js:110]: From the docs:
> Note: global version of the polyfill (dist/ResizeObserver.global) is deprecated and will be removed in the next major release.

Comment (dvoytenko) [reply build-system/tasks/update-packages.js:110]: Yep. I noticed it after the fact. So renamed it here.

**Comment by dvoytenko** [comment]
> nit: rename from `upgrade` to `fork` ?

done

**Review by samouri — APPROVED** [review]
Do you think we'd potentially be better served by promoting this to its own repo? E.g. `ampproject/resize-observer-polyfill`

**Comment by dvoytenko** [comment]
> Do you think we'd potentially be better served by promoting this to its own repo? E.g. `ampproject/resize-observer-polyfill`

We will do it if we do not get the response. But I think the rule of thumb for now is to give a few weeks before forking.

**Pull request merged**